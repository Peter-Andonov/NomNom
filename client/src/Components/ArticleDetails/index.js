import React, { useState, useEffect } from 'react';
import Axios from 'axios';
import styled from 'styled-components';
import { useParams } from 'react-router';
import { Editor, EditorState, convertFromRaw } from "draft-js";


const Wrapper = styled.div`
    padding: 3rem;
    position: absolute;
    top: 30vh;
    background-color: white;
    height: auto;
    width: 80%;
    display: flex;
    flex-direction: column;
    align-items: center;
`;

const EditorContainer = styled.div`
    margin-left: 5rem;
    width: 100%;
`;

const Image = styled.img`
    margin-bottom: 3rem;
    object-fit: contain;
`;

const MainTitle = styled.h1`
    margin: 0;
    margin-bottom: 2rem;
`;


const RecipeDetails = () => {

    const articleId = useParams();

    const [title, setTitle] = useState('');
    const [imageUrl, setImageUrl] = useState('');
    const [body, setBody] = useState(EditorState.createEmpty());

    useEffect(() => {
        const getArticle = async () => {

            const res = await Axios.get('http://localhost:5000/api/article', {
                params: {
                    id: articleId.id
                }
            });

            setTitle(res.data.title);

            setImageUrl(res.data.imageUrl);

            const bodyContentState = convertFromRaw(JSON.parse(res.data.body));
            setBody(EditorState.createWithContent(bodyContentState));
        };
        getArticle();
    }, []);

    return (
        <Wrapper>
            <MainTitle>{title}</MainTitle>
            <Image src={imageUrl} alt='Article' />
            <EditorContainer>
                <Editor editorState={body} readOnly={true} />
            </EditorContainer>
        </Wrapper>
    );
};


export default RecipeDetails;