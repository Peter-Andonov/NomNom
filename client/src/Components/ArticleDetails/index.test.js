import React from 'react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TestSetup from '../../Utils/TestUtils/test-utils';


import ArticleDetails from './index';


describe('ArticleDetails component', () => {
    it("does not render comment section or edit/delete buttons", () => {
        const { asFragment } = render(
            <TestSetup value={{
                loggedIn: false,
                user: null
            }}>
                <ArticleDetails />
            </TestSetup>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders comment section, but not edit/delete buttons", () => {
        const { asFragment } = render(
            <TestSetup value={{
                loggedIn: true,
                user: {
                    _id: '1234',
                    email: "john.doe@gmail.com",
                    role: 'user',
                    firstName: undefined,
                    lastName: undefined,
                    profilePicUrl: '',
                    favouriteRecipes: []
                }}}>
                <ArticleDetails />
            </TestSetup>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders comment section and admin edit/delete buttons", () => {
        const { asFragment } = render(
            <TestSetup value={{
                loggedIn: true,
                user: {
                    _id: '1234',
                    email: "john.doe@gmail.com",
                    role: 'admin',
                    firstName: undefined,
                    lastName: undefined,
                    profilePicUrl: '',
                    favouriteRecipes: []
                }}}>
                <ArticleDetails />
            </TestSetup>);
        expect(asFragment()).toMatchSnapshot();
    });
});