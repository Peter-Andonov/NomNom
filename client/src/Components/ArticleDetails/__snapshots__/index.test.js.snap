// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetails component does not render comment section or edit/delete buttons 1`] = `
<DocumentFragment>
  <div
    class="sc-fzoNJl kMRAUP"
  >
    <h1
      class="sc-fznxKY bLwHNe"
    />
    <img
      alt="Article"
      class="sc-fzpmMD hdaQQe"
      src=""
    />
    <div
      class="sc-fzoXWK gWfiXS"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-61n23"
            class="public-DraftEditor-content"
            contenteditable="false"
            spellcheck="false"
            style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="61n23"
                data-offset-key="6rkfp-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="6rkfp-0-0"
                >
                  <span
                    data-offset-key="6rkfp-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ArticleDetails component renders comment section and admin edit/delete buttons 1`] = `
<DocumentFragment>
  <div
    class="sc-fzoNJl kMRAUP"
  >
    <h1
      class="sc-fznxKY bLwHNe"
    />
    <img
      alt="Article"
      class="sc-fzpmMD hdaQQe"
      src=""
    />
    <div
      class="sc-fzoXWK gWfiXS"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-59qm8"
            class="public-DraftEditor-content"
            contenteditable="false"
            spellcheck="false"
            style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="59qm8"
                data-offset-key="cef1g-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="cef1g-0-0"
                >
                  <span
                    data-offset-key="cef1g-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-fzqARJ kcyFQq"
    >
      <button
        class="sc-fzqNqU gtBSgC"
      >
        Edit Article
      </button>
      <button
        class="sc-fzoyTs jJqnNu"
      >
        Delete Article
      </button>
    </div>
    <div
      class="sc-AxmLO fULdzK"
    >
      <div
        class="sc-fzozJi gAlura"
      >
        <div
          class="sc-fzoLsD bNKGwO"
        >
          Show comments
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ArticleDetails component renders comment section, but not edit/delete buttons 1`] = `
<DocumentFragment>
  <div
    class="sc-fzoNJl kMRAUP"
  >
    <h1
      class="sc-fznxKY bLwHNe"
    />
    <img
      alt="Article"
      class="sc-fzpmMD hdaQQe"
      src=""
    />
    <div
      class="sc-fzoXWK gWfiXS"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-8s7n5"
            class="public-DraftEditor-content"
            contenteditable="false"
            spellcheck="false"
            style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="8s7n5"
                data-offset-key="3c4cs-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="3c4cs-0-0"
                >
                  <span
                    data-offset-key="3c4cs-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-AxmLO fULdzK"
    >
      <div
        class="sc-fzozJi gAlura"
      >
        <div
          class="sc-fzoLsD bNKGwO"
        >
          Show comments
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
